common/reducers/wallet.ts

```diff
@@ -15,7 +15,7 @@ export interface State {
   // in ETH
   balance: Balance | { wei: null };
   tokens: {
-    [key: string]: TokenValue;
+    [key: string]: TokenValue | string;
```

Comment:
Why is there a string being added here? Everything should be a BN

Reply:
Instead of defaulting to a 0 value token on error, we could have a "Failed" string instead of trying to soft-fail. We can show the token value as "failed" in the UI to avoid confusion.

Reply:
I don't think we need to keep "Failed" in redux state -- we can simply display that in the UI based on some computation of other state